caravel/source_registry.py

```diff
@@ -20,3 +21,30 @@ def get_datasource(cls, datasource_type, datasource_id, session):
             .filter_by(id=datasource_id)
             .one()
         )
+
+    @classmethod
+    def get_datasource_by_name(cls, session, datasource_type, datasource_name,
+                               schema, database_name):
+        datasource_class = SourceRegistry.sources[datasource_type]
+        datasources = session.query(datasource_class).all()
```

Comment:
I wish that was that easy. I need to use properties for filtering, otherwise I have to implement 2 separate queries, as druid and table reference databases using different attributes.